pkg/nl/route.go

```diff
@@ -0,0 +1,199 @@
+package nl
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/telekom/das-schiff-network-operator/pkg/route"
+	schiff_unix "github.com/telekom/das-schiff-network-operator/pkg/unix"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+func GetProtocolName(p netlink.RouteProtocol) string {
+	protocol := p.String()
+	integer, err := strconv.Atoi(protocol)
+	if err != nil {
+		return protocol
+	}
+	switch integer {
+	case schiff_unix.RTPROT_NHRP:
+		return "nhrp"
+	case schiff_unix.RTPROT_LDP:
+		return "ldp"
+	case schiff_unix.RTPROT_SHARP:
+		return "sharp"
+	case schiff_unix.RTPROT_PBR:
+		return "pbr"
+	case schiff_unix.RTPROT_ZSTATIC:
+		return "zstatic"
+	case schiff_unix.RTPROT_OPENFABRIC:
+		return "openfabric"
+	case schiff_unix.RTPROT_SRTE:
+		return "srte"
+	case schiff_unix.RTPROT_COIL:
+		return "coil"
+	default:
+		return strconv.Itoa(int(p))
```

Comment:
Why not just return `protocol`? Seems like sort of a bold cast

Reply:
:+1: